src/dsql/DdlNodes.epp

```diff
@@ -1726,27 +1726,44 @@ void CreateAlterFunctionNode::execute(thread_db* tdbb, DsqlCompilerScratch* dsql
 	AutoSavePoint savePoint(tdbb, transaction);
 	bool altered = false;
 
+	bool alterIndividualParameters = !(body || external);
+
 	// first pass
 	if (alter)
 	{
-		if (executeAlter(tdbb, dsqlScratch, transaction, false, true))
+		if (alterIndividualParameters)
 		{
-			altered = true;
+			if (executeAlterIndividualParameters(tdbb, dsqlScratch, transaction, false, true))
+				altered = true;
+			else
+				status_exception::raise(Arg::Gds(isc_dyn_func_not_found) << Arg::Str(name));
 		}
 		else
 		{
-			if (create)	// create or alter
-				executeCreate(tdbb, dsqlScratch, transaction);
+			if (executeAlter(tdbb, dsqlScratch, transaction, false, true))
+			{
+				altered = true;
+			}
 			else
-				status_exception::raise(Arg::Gds(isc_dyn_func_not_found) << Arg::Str(name));
+			{
+				if (create)	// create or alter
+					executeCreate(tdbb, dsqlScratch, transaction);
+				else
+					status_exception::raise(Arg::Gds(isc_dyn_func_not_found) << Arg::Str(name));
+			}
```

Comment:
I believe this should finally look as `if ... else if ... else` rather than having a nested `if ... else`.